### DISCUSSION TITLE
transition-[background-color] transition-[color] is declared as a conflict of identical properties

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#12154] [Help] [answered]
The main problem is in the header, I'm also attaching screenshots. The problem is that when I specify two classes: `transition-[background-color]` and `transition-[color]`, the linter (?) underlines these properties and shows a message saying: `'transition-[background-color]' applies the same CSS properties as 'transition-[color]'.(cssConflict).` and vice versa.

These properties are analogous to CSS:
```CSS
transition-property: color, background-color
```
which is meant to animate different properties.

**Using:**

- VS Code: 1.83.0-insider
- tailwindcss: 3.3.3,
- postcss: 8.4.31,
- autoprefixer: 10.4.16

_Is this a bug or am I missing something?_

<img width="956" alt="Screenshot 2023-10-06 at 23 35 34" src="https://github.com/tailwindlabs/tailwindcss/assets/134743498/afe7f475-a887-4788-b2a7-413283a96701">
<img width="779" alt="Screenshot 2023-10-06 at 23 35 55" src="https://github.com/tailwindlabs/tailwindcss/assets/134743498/de56986f-962e-475c-a364-fe903fdd5c98">

**Comment by wongjn** [comment]
The warning is correct. This is because `transition-[color]` produces:

```css
.transition-\[color\] {
  transition-property: color;
  /* … */
}
```

And `transition-[background-color]` produces:

```css
.transition-\[background-color\] {
  transition-property: background-color;
  /* … */
}
```

So if you look at these two rules, the `transition-property` would not combine – one would override the other and this is what the warning is for.

If you are after both `background-color` and `color` being transitioned, you'd want to look at using either `transition`, `transition-colors` or `transition-[background-color,color]`

**Reply by ghost** [reply]
Oh, I get it. But I think it's still an oversight, because the scenario where you list a list of properties you want to change is quite popular, like this property:

```css
transition-property: color, background-color, border-color
```

Ideally it would be cool to have a property like: `transition-[color background-color border-color]` that groups them together. But I don't know, can the authors add support for this or do they have other plans?

Thanks for the answer anyway @wongjn

**Reply by wongjn** [reply, marked as answer]
As per my answer above, you could use:

- `transition` which applies
  ```
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;
  ```
- `transition-colors` which applies
  ```
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke
  ```
- `transition-all` which applies
  ```
  transition-property: all
  ```
- `transition-[background-color,color,border-color]` which applies
  ```
  transition-property: background-color,color,border-color
  ```
